packages/serialize/types/index.d.ts

```diff
@@ -56,17 +59,18 @@ export type Keyframes = {
   toString: () => string
 } & string
 
-export interface ArrayInterpolation<MP> extends Array<Interpolation<MP>> {}
-export interface ObjectInterpolation<MP>
+export interface ArrayInterpolation<MP = { theme: Theme }>
+  extends Array<Interpolation<MP>> {}
+export interface ObjectInterpolation<MP = { theme: Theme }>
   extends CSSPropertiesWithMultiValues,
     CSSPseudos<MP>,
     CSSOthersObject<MP> {}
 
-export interface FunctionInterpolation<MergedProps> {
+export interface FunctionInterpolation<MergedProps = { theme: Theme }> {
   (mergedProps: MergedProps): Interpolation<MergedProps>
 }
 
-export type Interpolation<MergedProps = undefined> =
+export type Interpolation<MergedProps = { theme: Theme }> =
```

Comment:
actually I think that `Interpolation` is also used in `css` definition and in there you don't have access to `Theme`, so the previous version with default being `undefined` was correct for `css`'s case

Reply:
Yeah, this and the `serializeStyles` stuff should be better viewed by @JakeGinnivan. Not on the same train of thought as he is on with these.

I also invited you @JakeGinnivan as a member to my repo, so it's not mandatory to have to PR.

Reply:
> > css definition and in there you don't have access to Theme

> Just for my understanding, why is this the case?

I've meant `css` function - you can't do `css(() => /**/)`. From what remember such function just gets stringified. `css` call is being executed without any notion of context, you can hoist it wherever you want etc - so it has no access to theme or props. 

A different story is the `css` props - where you can get access to the builtin Theme.

> Looking at the other types elsewhere we could actually do another round of simplification if we could add theme into these types directly

What kind of simplification do you have in mind? I've noticed now that we have Interpolation, CSSInterpolation and even InterpolationWithTheme and I think they might be used sometimes in wrong contexts (especially the Interpolation type). I need to draw out the dependency tree of those types somewhere for myself to analyze this further 😅